spec/system/admin/subscriptions_spec.rb

```diff
@@ -357,6 +357,23 @@ def open_subscription_products_panel
               expect(page).to have_selector 'td.total', text: "$27.50"
             end
           end
+
+          context 'and deleting the existing product' do
+            before do
+              within 'table#subscription-line-items tr.item', match: :first do
+                find("a.delete-item").click
+              end
+            end
+
+            it 'has content Please add at least one product and subscription does not change' do
+              click_button('Next')
+
+              expect{
+                click_button('Create Subscription')
+                expect(page).to have_content 'Please add at least one product'
+              }.to_not change(Subscription, :count)
+            end
+          end
```

Comment:
Hmm, so the old test was are adding a product, deleting it, and testing what happens when you try to proceed without a product. I think we should instead arrange it this way:

```ruby
# without a product
it 'has content Please add at least one product and subscription does not change'

context 'with a product' do
  it 'has description, price estimates, quantity and total'

  context 'and deleting the existing product' do
    it 'deletes the product' # a new test
    # it 'can still add another new product' # technically this was implicitly tested before, but I think unnecessary to test
  end
end


```

Reply:
@dacook this is the only one I was not comfortable.

> without a product
> it 'has content Please add at least one product and subscription does not change'

It is the first scenario above this context where it is: `and click next button without adding at least one product`.

I tried to check inside it the Create Subscription button event, But it is shown on the next page and we can go to the next page only with a product. The error was:

`Capybara::ElementNotFound:
       Unable to find visible button "Create Subscription" that is not disabled`.

Is it possible to start it by trying to create a subscription without a product on the create subscription page?

If I went to the wrong conclusion based on what you wrote, please, let me know.

Reply:
Hi Paulo Felipe, I'm so sorry to lead you down the wrong path. I didn't realise this, so thank you for pointing it out.